stats/view/aggregation_data.go

```diff
@@ -17,101 +17,63 @@ package view
 
 import (
 	"math"
+
+	"go.opencensus.io/exporter"
 )
 
-// AggregationData represents an aggregated value from a collection.
-// They are reported on the view data during exporting.
-// Mosts users won't directly access aggregration data.
-type AggregationData interface {
-	isAggregationData() bool
+// aggregator receives data points and aggregates them in place.
+type aggregator interface {
 	addSample(v float64)
-	clone() AggregationData
-	equal(other AggregationData) bool
+	writeTo(d *exporter.AggregationData)
 }
 
-const epsilon = 1e-9
+// TODO(ramonza): remove all the aggregator types and replace with just a
+// func(*exporter.AggregationData) that updates its argument in-place.
 
-// CountData is the aggregated data for the Count aggregation.
-// A count aggregation processes data and counts the recordings.
-//
-// Most users won't directly access count data.
-type CountData int64
+type countData int64
 
-func newCountData(v int64) *CountData {
-	tmp := CountData(v)
+func newCountData(v int64) *countData {
+	tmp := countData(v)
 	return &tmp
 }
 
-func (a *CountData) isAggregationData() bool { return true }
-
-func (a *CountData) addSample(_ float64) {
+func (a *countData) addSample(_ float64) {
 	*a = *a + 1
 }
 
-func (a *CountData) clone() AggregationData {
-	return newCountData(int64(*a))
-}
-
-func (a *CountData) equal(other AggregationData) bool {
-	a2, ok := other.(*CountData)
-	if !ok {
-		return false
-	}
-
-	return int64(*a) == int64(*a2)
+func (a *countData) writeTo(dd *exporter.AggregationData) {
```

Comment:
Perhaps `copyTo`, `write` is a sacred prefix in Go that implies an actual `[]byte` change somewhere.